plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -234,23 +234,22 @@ public static ReaderPageSource createPageSource(
             messageColumn = getColumnIO(fileSchema, requestedSchema);
 
             Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            List<TupleDomain<ColumnDescriptor>> parquetTupleDomains;
-            List<TupleDomainParquetPredicate> parquetPredicates;
-            if (options.isIgnoreStatistics()) {
-                parquetTupleDomains = ImmutableList.of();
-                parquetPredicates = ImmutableList.of();
-            }
-            else {
-                ImmutableList.Builder<TupleDomain<ColumnDescriptor>> parquetTupleDomainsBuilder = ImmutableList.builderWithExpectedSize(disjunctTupleDomains.size());
-                ImmutableList.Builder<TupleDomainParquetPredicate> parquetPredicatesBuilder = ImmutableList.builderWithExpectedSize(disjunctTupleDomains.size());
-                for (TupleDomain<HiveColumnHandle> tupleDomain : disjunctTupleDomains) {
+
+            ImmutableList.Builder<TupleDomain<ColumnDescriptor>> parquetTupleDomainsBuilder = ImmutableList.builderWithExpectedSize(disjunctTupleDomains.size());
+            ImmutableList.Builder<TupleDomainParquetPredicate> parquetPredicatesBuilder = ImmutableList.builderWithExpectedSize(disjunctTupleDomains.size());
+            for (TupleDomain<HiveColumnHandle> tupleDomain : disjunctTupleDomains) {
+                if (options.isIgnoreStatistics()) {
+                    parquetTupleDomainsBuilder.add(TupleDomain.all());
+                    parquetPredicatesBuilder.add(buildPredicate(requestedSchema, TupleDomain.all(), descriptorsByPath, timeZone));
+                }
+                else {
                     TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, tupleDomain, fileSchema, useColumnNames);
                     parquetTupleDomainsBuilder.add(parquetTupleDomain);
                     parquetPredicatesBuilder.add(buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone));
                 }
-                parquetTupleDomains = parquetTupleDomainsBuilder.build();
-                parquetPredicates = parquetPredicatesBuilder.build();
             }
+            List<TupleDomain<ColumnDescriptor>> parquetTupleDomains = parquetTupleDomainsBuilder.build();
```

Comment:
if ignore-stats, this fills parquetTupleDomains with a series of TupleDomain.all objects, which is not very useful.

ionstead of this, let's change `for (int i = 0; i < disjunctTupleDomains.size(); i++) {` loop iteration to iterate over `parquetTupleDomains.size()` instead iof `disjunctTupleDomains.size()`

Reply:
That will skip the following `blocks.add` and `blockStarts.add`. Is that OK?

Reply:
@findinpath please chime in

Reply:
@findepi i agree with your suggestion.
Adapting the loop to use `parquetTupleDomains.size()` instead of `disjunctTupleDomains.size()` should give us the same result - ALL the blocks in the file.
@jinyangli34  the loop https://github.com/trinodb/trino/pull/19761/files#diff-bb16d0894036f6a50fd882558df497a4ddbf2907097b8a33118c4f6ce147753dR240  is to be used only if `options.isIgnoreStatistics()` is `false`.


> That will skip the following `blocks.add` and `blockStarts.add`
This happens now as well right ? 
https://github.com/trinodb/trino/blob/e9ba634097147e97bdd19a84f9150685ae760c59/plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java#L281

Reply:
Hi @findinpath @findepi, to make sure I understand correctly, parquetTupleDomains will still hold TupleDomain.all objects, not an empty list. Is that correct?